### DISCUSSION TITLE
Incorporate GraphServiceClient into MicrosoftGraphProvider

### DISCUSSION THREAD

**Discussion by bgiromini** [ServiceStack/Discuss#29] [Ideas]
It would be nice to be able to access other features of the Microsoft Graph API, like if an instance of the GraphServiceClient was made available. This would allow users to use the typed client to send emails, create tasks, etc.

**Comment by Layoric** [comment]
Currently ServiceStack doesn't take a dependency on the `Microsoft.Graph` NuGet package so there isn't any `GraphServiceClient` to expose. 

This library should be able to be used with the [`DelegateAuthenticationProvider`](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/overview.md#delegateauthenticationprovider) once related access key has been obtained if you are already using the `MicrosoftGraphAuthProvider` `AccessTokenUrl ` or by overriding `CreateAuthInfoAsync` in your own `OAuth2Provider`.

Could you give an example of how you would want to use the current `MicrosoftGraphAuthProvider` with the additional API functionality from the Microsoft Graph API or sample code of how you might expect the API to look when using with ServiceStack?

**Comment by bgiromini** [comment]
At the time I opened this issue, I did not know about DelegateAuthenticationProvider so I thought getting it incorporated into the base code was the solution. I have since learned that the access token is stored with the session so using that with the DelegateAuthenticationProvider has solved my problem.
